Sources/StreamChatUI/Components.swift

```diff
@@ -82,7 +82,54 @@ public struct _Components<ExtraData: ExtraDataTypes> {
     /// The injector used for injecting file attachment views
     public var filesAttachmentInjector: _FilesAttachmentViewInjector<ExtraData>.Type = _FilesAttachmentViewInjector<ExtraData>.self
 
-    public var channelList = ChannelList()
+    // MARK: - Channel list components
+
+    /// The logic to generate a name for the given channel.
+    public var channelNamer: ChatChannelNamer<ExtraData> = DefaultChatChannelNamer()
+
+    /// The collection view of the channel list.
+    public var channelListCollectionView: UICollectionView.Type = UICollectionView.self
+
+    /// The collection view layout of the channel list.
+    public var channelListLayout: UICollectionViewLayout.Type = ListCollectionViewLayout.self
+
+    /// The `UICollectionViewCell` subclass that shows channel information.
+    public var channelCell: _ChatChannelListCollectionViewCell<ExtraData>.Type =
+        _ChatChannelListCollectionViewCell<ExtraData>.self
+
+    /// The channel cell separator in the channel list.
+    public var channelCellSeparator: UICollectionReusableView.Type = CellSeparatorReusableView.self
+
+    /// The view in the channel cell that shows channel actions on swipe.
+    public var channelActionsView: _SwipeableView<ExtraData>.Type =
+        _SwipeableView<ExtraData>.self
+
+    /// The `UIStackView` that arranges buttons revealed by swipe gesture in `channelActionsView`.
+    public var channelActionsStackView: UIStackView.Type = UIStackView.self
+
+    /// The view that shows channel information.
+    public var channelContentView: _ChatChannelListItemView<ExtraData>.Type = _ChatChannelListItemView<ExtraData>.self
+
+    /// The label that shows channel's title (the result of `ChatChannelNamer`).
+    public var channelTitleLabel: UILabel.Type = UILabel.self
+
+    /// The label that shows a subtitle of channel, typically shows if users are typing or last message,
+    /// see `typingMemberOrLastMessageString` in `ChatChannelListItemView`.
+    public var channelSubtitleLabel: UILabel.Type = UILabel.self
+
+    /// The label that shows the timestamp of the latest channel message.
+    public var channelTimestampLabel: UILabel.Type = UILabel.self
```

Comment:
Just moved it from `Component.ChannelList` but do we need to expose these ones? If we do I'd consider injection via `ChatChannelListItemView`